capstone/src/main/java/com/google/univiz/scorecard/CollegeDataApiImpl.java

```diff
@@ -0,0 +1,37 @@
+package com.google.univiz.scorecard;
+
+import com.google.gson.Gson;
+import com.google.univiz.CollegeData;
+import com.google.univiz.CollegeDataApi;
+import com.google.univiz.api.CollegeId;
+import java.util.ArrayList;
+import java.util.List;
+import javax.inject.Inject;
+
+public class CollegeDataApiImpl implements CollegeDataApi {
+  private final ScorecardConverter scorecardConverter;
+  private final CollegeIdReaderProvider readerProvider;
+
+  @Inject
+  public CollegeDataApiImpl(
+      ScorecardConverter scorecardConverter, CollegeIdReaderProvider readerProvider) {
+    this.scorecardConverter = scorecardConverter;
+    this.readerProvider = readerProvider;
+  }
+
+  @Override
+  public List<CollegeData> getCollegesById(List<CollegeId> ids) {
+    List<CollegeData> colleges = new ArrayList<>();
+    for (CollegeId id : ids) {
+      // Then deserialize into a ScorecardResponse
+      Gson gson = new Gson();
```

Comment:
I think your issue is that Gson doesn't know about the autovalue types.
```suggestion
      Gson gson = new GsonBuilder().registerTypeAdapterFactory(GenerateTypeAdapter.FACTORY).create();
```